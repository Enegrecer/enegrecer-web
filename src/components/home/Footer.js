import React from 'react';
import PropTypes from 'prop-types';

const Footer = props => (
  <div className="en-footer">
    <div className="container">
      {props.children}
    </div>
  </div>
);

Footer.propTypes = {
  children: PropTypes.element.isRequired,
};

export default Footer;
