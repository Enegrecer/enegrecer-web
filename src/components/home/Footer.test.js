import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Footer from './Footer';

describe('Footer', () => {
  const wrapper = shallow(
    <Footer>
      <div id="footer-content" />
    </Footer>,
  );

  it('possui a mesma estrutura', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('tem a classe en-footer', () => {
    expect(wrapper.hasClass('en-footer')).toBeTruthy();
  });

  it('tem um container', () => {
    expect(wrapper.find('.container').exists()).toBeTruthy();
  });

  it('tem o filho #footer-content', () => {
    expect(wrapper.find('#footer-content').exists()).toBeTruthy();
  });
});
