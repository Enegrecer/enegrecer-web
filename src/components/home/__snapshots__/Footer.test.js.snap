// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer possui a mesma estrutura 1`] = `
<div
  className="en-footer"
>
  <div
    className="container"
  >
    <div
      id="footer-content"
    />
  </div>
</div>
`;
